sbin/common/common.sh

```diff
@@ -18,6 +18,19 @@
 # shellcheck disable=SC2153
 function setOpenJdkVersion() {
   local forest_name=$1
+set -x
+  if [ -e "${forest_name}" ] ; then
```

Comment:
There are two things  mixed togehter.

Yyou are absolutly correct and `--version` is exactly what OPENJDK_FOREST_ALTID was substituing. I will remove `--jdk-dir-altid` in favour of `--version`  It will need some more changes, nut better then additional single purposed switch. Sorry for missing `--version`.

I woudl very like to avoid the `--openjdk-source`. it would kill the script logic. Teh script expects repository_name or dir/tarball. The  usage of `make_any_jdk.sh dir` is much more strigthforward and much usable then the verbose one. In additoon, the script will be pulluted by nasty (imo) if:

"expects exactly one argument, unless --openjdk-source si used, then it expects no argument"

Unless you are expecting the usage as `make_any_jdk.sh  `--openjdk-source some_terrible_name/path/zip   id`
Where id is still mandatory, and is providing the necessary "original repo" information.
I like "my" approach a bti more, because it have less mandatory arguments
 - `make_any_jdk.sh  some_terrible_name/path/zip   ` +optional `version id` - because the name/path/zip maybe akready proper version.

In comaprison to
 - `make_any_jdk.sh  --openjdk-source some_terrible_name/path/zip   ` with mandatory  --openjdk-source  and otional short id.


wdyt?

Reply:
So maybe, the syntax of makejdk-any-platform.sh, is:
```
makejdk-any-platform.sh <jdk_version>
```
Where jdk_version is what has historically been named as the "forest version"
From jdk_version you must be able to set:
```
BUILD_CONFIG[OPENJDK_CORE_VERSION]       eg. jdk21
BUILD_CONFIG[OPENJDK_FOREST_NAME]        eg. jdk21u
BUILD_CONFIG[OPENJDK_FEATURE_NUMBER] eg. 21
```
These are needed throughtout the sbin/build.sh build logic.. So you will need to set these from some_terrible_name/path/zip ??

Reply:
I would actually prefer:
```
makejdk-any-platform.sh jdk21u --openjdk-source some_terrible_name/path/zip
```
because the existing logic very clearly set:
```
BUILD_CONFIG[OPENJDK_CORE_VERSION]       eg. jdk21
BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]   eg. 21
```

Reply:
> So maybe, the syntax of makejdk-any-platform.sh, is:
> 
> ```
> makejdk-any-platform.sh <jdk_version>
> ```
> 
> Where jdk_version is what has historically been named as the "forest version" From jdk_version you must be able to set:
> 
> ```
> BUILD_CONFIG[OPENJDK_CORE_VERSION]       eg. jdk21
> BUILD_CONFIG[OPENJDK_FOREST_NAME]        eg. jdk21u
> BUILD_CONFIG[OPENJDK_FEATURE_NUMBER] eg. 21
> ```
> 
> These are needed throughtout the sbin/build.sh build logic.. So you will need to set these from some_terrible_name/path/zip ??

Exactly, And thats what that dir/zip i susually named. See https://github.com/adoptium/temurin-build/pull/3727
For cases Where the the dir/tarball is nonsense from this pint of vie, I introduced that `altid`, which will be substituted by `version`.  And used that to set featureversion, if used.

Reply:
> I would actually prefer:
> 
> ```
> makejdk-any-platform.sh jdk21u --openjdk-source some_terrible_name/path/zip
> ```
> 
> because the existing logic very clearly set:
> 
> ```
> BUILD_CONFIG[OPENJDK_CORE_VERSION]       eg. jdk21
> BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]   eg. 21
> ```

Probably my main reason is to save typing `--openjdk-source`  every time I wish to build some properly  named directory:(   I was resuing the logic which set
> ```
> BUILD_CONFIG[OPENJDK_CORE_VERSION]       eg. jdk21
> BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]   eg. 21
> ```

properly. Do you really insists?